server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStoreChoosingConfiguration.java

```diff
@@ -0,0 +1,80 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.api;
+
+import java.util.stream.Stream;
+
+import org.apache.commons.configuration.Configuration;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class BlobStoreChoosingConfiguration {
+
+    public enum BlobStoreName {
+        CASSANDRA("cassandra"),
+        SWIFT("swift");
+
+        static BlobStoreName from(String name) {
+            return Stream.of(values())
+                .filter(blobName -> blobName.getName().equals(name))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException(String.format("%s is not a valid name of BlobStores", name)));
+        }
+
+        private final String name;
+
+        BlobStoreName(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    static final String DEFAULT_BLOBSTORE_PROPERTY = "objectstore.defaultStorage";
```

Comment:
`defaultStorage` seems not like a good name.

It is not a storage by default that we specify but THE storage to vbe used.

Let me propose : `objectstore.implementation`